packages/serverless-contracts/src/contracts/eventBridge/features/putEvents.ts

```diff
@@ -0,0 +1,137 @@
+import {
+  PutEventsCommand,
+  PutEventsRequestEntry,
+  PutEventsResultEntry,
+} from '@aws-sdk/client-eventbridge';
+import _PQueue from 'p-queue';
+
+import { chunkEntriesBatch } from '../../../utils';
+import { fixESMInteropIssue } from '../../../utils/fixESMInteropIssue';
+import { EventBridgeContract } from '../eventBridgeContract';
+import { PutEventsBuilderArgs, PutEventsSideEffect } from '../types/putEvents';
+
+const PQueue = fixESMInteropIssue(_PQueue);
+
+const MAX_BATCH_SIZE = 256_000; // 256Kb
+const MAX_BATCH_LENGTH = 10;
+
+const defaultOptions = {
+  throughputCallsPerSecond: 400, // Minial default throughput. Some regions can have up to 10_000. See: https://docs.aws.amazon.com/eventbridge/latest/userguide/eb-quota.html
+  throwOnFailure: true,
+};
+
+const getThroughputQueueConfig = (
+  throughputCallsPerSecond: number,
+): { intervalCap: number; interval: number } => ({
+  intervalCap: throughputCallsPerSecond,
+  interval: 1000,
+});
+
+const computeEventSize = (event: PutEventsRequestEntry): number => {
```

Comment:
Can you add a bit of JSDoc to explain why this is needed and the approximations you used?

Reply:
I don't have lots of input, I copy-pasted this from https://github.com/fredericbarthelet/typebridge/blob/master/src/Bus.ts